webapp/src/containers/Rhs/projectDetails/index.tsx

```diff
@@ -26,31 +30,61 @@ const data: SubscriptionDetails[] = [
     },
 ];
 
-type ProjectDetailsProps = {
-    title: string
-}
+const ProjectDetails = (projectDetails: ProjectDetails) => {
+    // State variables
+    const [showConfirmationModal, setShowConfirmationModal] = useState(false);
 
-const ProjectDetails = ({title}: ProjectDetailsProps) => {
+    // Hooks
     const dispatch = useDispatch();
+    const usePlugin = usePluginApi();
 
     const handleResetProjectDetails = () => {
         dispatch(resetProjectDetails());
     };
 
+    /**
+     * Opens a confirmation modal to confirm unlinking a project
+     */
+    const handleUnlinkProject = () => {
+        setShowConfirmationModal(true);
+    };
+
+    // Handles unlinking a project
+    const handleConfirmUnlinkProject = async () => {
+        const unlinkProjectStatus = await usePlugin.makeApiRequest(plugin_constants.pluginApiServiceConfigs.unlinkProject.apiServiceName, projectDetails);
+
+        if (unlinkProjectStatus) {
+            handleResetProjectDetails();
+            setShowConfirmationModal(false);
+        }
+    };
+
     // Reset the state when the component is unmounted
     useEffect(() => {
-        return handleResetProjectDetails();
+        return () => {
+            handleResetProjectDetails();
+        };
     }, []);
 
     return (
         <>
+            <ConfirmationModal
+                isOpen={showConfirmationModal}
+                onHide={() => setShowConfirmationModal(false)}
+                onConfirm={handleConfirmUnlinkProject}
+                isLoading={usePlugin.getApiState(plugin_constants.pluginApiServiceConfigs.unlinkProject.apiServiceName, projectDetails).isLoading}
+                confirmBtnText='Unlink'
+                description={`Are you sure you want to unlink ${projectDetails?.projectName}?`}
+                title='Confirm Project Unlink'
+            />
             <BackButton onClick={handleResetProjectDetails}/>
             <div className='d-flex'>
-                <p className='rhs-title'>{title}</p>
+                <p className='rhs-title'>{projectDetails.projectName}</p>
                 <IconButton
                     tooltipText='Unlink project'
                     iconClassName='fa fa-chain-broken'
                     extraClass='project-details-unlink-button unlink-button'
+                    onClick={() => handleUnlinkProject()}
```

Comment:
```suggestion
                    onClick={handleUnlinkProject}
```

Reply:
this has been fixed during refactoring in later PR